test/IISIntegration.FunctionalTests/Utilities/IISFunctionalTestBase.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Xunit.Abstractions;
+
+namespace IISIntegration.FunctionalTests.Utilities
+{
+    public class IISFunctionalTestBase : FunctionalTestsBase
+    {
+        public IISFunctionalTestBase(ITestOutputHelper output = null) : base(output)
+        {
+        }
+
+        protected override Task<IISDeploymentResult> DeployAsync(DeploymentParameters parameters)
+        {
+            if (parameters.SiteName == null)
+            {
+                parameters.SiteName = "HttpTestSite";
```

Comment:
https://github.com/aspnet/Hosting/issues/1423 would get rid of both of these.

Reply:
We need transformation callbacks in addition to it.